### PR TITLE
Add bastion host security

### PR DESCRIPTION
**Pull request by robinsturmeit**
This commit is quite extensive. It aims to ensure that there is no
access to the cluster from the public internet when a bastion host is
used. In addition, there were some usability improvements. Quick
overview on the changes:

- Use only a private endpoint in EKS when a bastion is used
  The main and major change, ensuring that the bastion host is the only
  way to work on the k8s cluster. This necessitates some acrobatics,
  though: As the EKS module cannot configure the cluster itself after it
  has been set up, the user must provide a role that is used to set up the
  cluster. This role is assumed by the bastion host during startup and the
  bastion host then inserts its own instance profile as cluster admin.

- Add some QOL improvements to the bastion: Aliasing `kubectl` with `k`,
  get current EKS kubeconfig automatically when starting a new session, enable
  `kubectl` autocompletion, ensure that AWS CLI is setup to use the instance
  profile, option to add additional tags to the bastion.

- Add option to use EC2 Instance Connect. Similar to use using OSLogin
  in GKE, this commit adds the option to use EC2 Instance Connect and
  does so by default. The main advantage is that AWS will manage SSH
  keys for you and bastion host access can be administered using IAM
  users, roles and permissions. This has also been implemented for SSH
  connections from the bastion host to the Nomad clients.

- Add lots of documentation and improve formatting

### CODE REVIEW

**Comment by ryanwohara** [comment]
This is a lot to review but at a first glance I like the approach

File: eks/outputs.tf [hunk 1]
```diff
@@ -31,3 +31,13 @@ output "nomad_server_key" {
 output "nomad_tls_ca" {
   value = module.nomad.nomad_tls_ca
 }
+
+locals {
+  bastion_connect_command = var.bastion_key == "" ? "mssh ubuntu@${aws_instance.bastion[0.0].id}" : "ssh ubuntu@${aws_instance.bastion[0.0].public_ip}"
```
Comment (ryanwohara) [inline eks/outputs.tf:36]: Should both sides of this if statement use the same command?

**Review by ryanwohara — CHANGES_REQUESTED** [review]
It seems like we're moving to mssh but I've never heard of it and it is not installed locally. It does not seem like a good idea to implement another tool.
Additionally the bastion access instructions mention an issue with -L but does not provide an example.

File: eks/outputs.tf (hunk 1 above)
Comment (robinsturmeit) [reply eks/outputs.tf:36]: They don't: One side uses `mssh` with the `instance-id` and the other side uses `ssh` with the public IP

**Comment by robinsturmeit** [comment]
> It seems like we're moving to mssh but I've never heard of it and it is not installed locally. It does not seem like a good idea to implement another tool.
> Additionally the bastion access instructions mention an issue with -L but does not provide an example.

It's unfortunate that aws hasn't included their ssh wrapper into their main CLI like Google did with `gcloud compute ssh` – however, I do document how to use `ssh` in connection with pushing the ssh key first. I will add an example for the -L flag in ssh.

**Comment by robinsturmeit** [comment]
As for the failed terraform fmt: There seems to be a change in style between my local terraform version and the version we are using in the pipeline 🤔

**Comment by robinsturmeit** [comment]
Closing this as bastion host support will be removed

**Pull request closed without merging**